homeassistant/components/group/sensor.py

```diff
@@ -0,0 +1,411 @@
+"""This platform allows several sensors to be grouped into one sensor to provide numeric combinations."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from datetime import datetime
+import logging
+import statistics
+from typing import TYPE_CHECKING, Any
+
+import voluptuous as vol
+
+from homeassistant.components.input_number import DOMAIN as INPUT_NUMBER_DOMAIN
+from homeassistant.components.number import DOMAIN as NUMBER_DOMAIN
+from homeassistant.components.sensor import (
+    CONF_STATE_CLASS,
+    DEVICE_CLASSES_SCHEMA,
+    DOMAIN,
+    PLATFORM_SCHEMA as PARENT_PLATFORM_SCHEMA,
+    STATE_CLASSES_SCHEMA,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_DEVICE_CLASS,
+    CONF_ENTITIES,
+    CONF_NAME,
+    CONF_TYPE,
+    CONF_UNIQUE_ID,
+    CONF_UNIT_OF_MEASUREMENT,
+    STATE_UNAVAILABLE,
+    STATE_UNKNOWN,
+)
+from homeassistant.core import Event, HomeAssistant, State, callback
+from homeassistant.helpers import config_validation as cv, entity_registry as er
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_track_state_change_event
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType, StateType
+
+from . import GroupEntity
+
+DEFAULT_NAME = "Sensor Group"
+CONF_ALL = "all"
+CONF_ROUND_DIGITS = "round_digits"
+
+ATTR_MIN_VALUE = "min_value"
+ATTR_MIN_ENTITY_ID = "min_entity_id"
+ATTR_MAX_VALUE = "max_value"
+ATTR_MAX_ENTITY_ID = "max_entity_id"
+ATTR_MEAN = "mean"
+ATTR_MEDIAN = "median"
+ATTR_LAST = "last"
+ATTR_LAST_ENTITY_ID = "last_entity_id"
+ATTR_RANGE = "range"
+ATTR_SUM = "sum"
+SENSOR_TYPES = {
+    ATTR_MIN_VALUE: "min",
+    ATTR_MAX_VALUE: "max",
+    ATTR_MEAN: "mean",
+    ATTR_MEDIAN: "median",
+    ATTR_LAST: "last",
+    ATTR_RANGE: "range",
+    ATTR_SUM: "sum",
+}
+SENSOR_TYPE_TO_ATTR = {v: k for k, v in SENSOR_TYPES.items()}
+
+# No limit on parallel updates to enable a group calling another group
+PARALLEL_UPDATES = 0
+
+PLATFORM_SCHEMA = PARENT_PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_ENTITIES): cv.entities_domain(
+            [DOMAIN, NUMBER_DOMAIN, INPUT_NUMBER_DOMAIN]
+        ),
+        vol.Required(CONF_TYPE): vol.All(cv.string, vol.In(SENSOR_TYPES.values())),
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_UNIQUE_ID): cv.string,
+        vol.Optional(CONF_ALL, default=False): cv.boolean,
```

Comment:
Let's change this to `ignore non-numeric` to not confuse it with `all` which is supported by `binary_sensor`, `light` and `switch` groups.